### ISSUE TITLE
Install/Update fails at very end as it is about to start the update for macOS - Error Timeout Reached

### ISSUE THREAD

**Issue by dvasquezavant** [grahampugh/erase-install#320]
**Describe the bug**
I am in the process of updating 12.6 > 12.6.1 on x86 and arm64 laptops and on some I am seeing issues where it gets to the very end and then fails or files out and the error message that is provided is vague. Recently I was told to use the --update and this will verify if there is a current install then if it is valid it will use it or overwrite it. 

![image](https://user-images.githubusercontent.com/101146428/204911760-37abb988-33a4-4acc-9e68-1879cff95041.png)

**To Reproduce**
 - If applicable, please provide the full command used when running the script.
 Full command:
 
![image](https://user-images.githubusercontent.com/101146428/204911862-bb84c413-3f0f-48c3-a08c-6d32715f2fde.png)

 - Please describe how the script was run. [e.g. Standalone via Terminal, in Jamf Policy from the uploaded pkg, in Jamf policy from uploaded script only, via another MDM provider]
 **The script is being run from Jamf Pro as a policy. and set to run 3 times if it fails**

![Screenshot 2022-11-30 at 4 06 57 PM](https://user-images.githubusercontent.com/101146428/204918444-5b8f7f3b-a538-4980-a76a-a68f3dc94bd7.png)

**Expected behavior**
I am expecting the upgrade process to finish successfully. I would also like to know why it might be failing with a little more clarity.
**I am not saying the logs are not useful here, not at all, I am saying that all looks to be going well then it simply times out and fails at the end.**

**Code/log output**
Please supply the full command used, and if applicable, add full output from Terminal or from `/var/log/erase-install.log`. Either upload the log or paste the output in a code block (triple backticks at the start and end of the code block please!).

**Screenshots**
![Screenshot 2022-11-30 at 3 15 26 PM](https://user-images.githubusercontent.com/101146428/204916645-4844f385-a7f0-4c3e-b102-b2b6a23feb56.png)
![Screenshot 2022-11-30 at 3 16 40 PM](https://user-images.githubusercontent.com/101146428/204916671-189dd3de-c4e2-4557-a1fc-b4fb10289ec9.png)

![Screenshot 2022-11-30 at 3 16 51 PM](https://user-images.githubusercontent.com/101146428/204916682-2ce36d99-017e-4b6e-b41f-07e0b368fa07.png)

**Environment (please complete the following information):**
 - OS version [e.g. 12.0.1] = **12.6.0**
 - erase-install version [e.g. 24.0] - please upgrade to the latest version before submitting a bug report. **Latest version 27.1**

**Additional context**
I see the process start, I see in the logs that the end user confirms the install, and the password for the volume owner is supplied and accepted. The laptop has adequate storage and is connected to power.  
I also truly appreciate your help and dedication to this process as it has helped me and my organization upgrade many laptops. 
I am wanting to figure out if this error or the error is due to a corrupt downloaded file if the laptop goes to sleep and if the commands to **startosinstall** are failing.  
![Xnip2022-11-30_15-19-00](https://user-images.githubusercontent.com/101146428/204917599-57aacbf3-b378-4fad-b805-24068c50772a.jpg)
![Xnip2022-11-30_15-20-12](https://user-im
![Xnip2022-11-30_15-21-18](https://user-images.githubusercontent.com/101146428/204917610-ab506b23-21cd-4851-9db9-78727f045b11.jpg)
ages.githubusercontent.com/101146428/204917606-9dd99e8c-4777-4966-a8c5-abc5f683b8e8.jpg)

I think is also important to note that in testing I am not seeing this issue on most of my test pilot laptops and team. 

The PID does vary at the end of the run if the process does fail. 

![Screenshot 2022-11-30 at 4 06 14 PM](https://user-images.githubusercontent.com/101146428/204918336-73d8b9f9-9bbd-40fd-9065-8b2caa2c9c8f.png)

**Comment by grahampugh** [comment]
Something is causing that installer to fail to prepare properly. I would recommend clearing it and downloading again. I think the timeout just got reached after the preparation already failed.

**Comment by dvasquezavant** [comment]
Thank you for getting back to me!

I have been cleaning them out, per my last ticket with you and that did
help. I also added the *--update* command to over right the file if it
exists.

I am finding that removing the --*move* command cuts down on the time and
work the process has to complete.

Hard to tell what exactly is going on and if it is a power issue, sleep
issue, or the laptop lid is being closed or if it fails. As you can see
from the logs the process looks to be going very smoothly then boon, fail
with that time out.

Again thank you for the time to get back to me!
Dom



On Tue, Jan 31, 2023 at 2:55 PM Graham Pugh ***@***.***>
wrote:

> Something is causing that installer to fail to prepare properly. I would
> recommend clearing it and downloading again. I think the timeout just got
> reached after the preparation already failed.
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_grahampugh_erase-2Dinstall_issues_320-23issuecomment-2D1411057408&d=DwMCaQ&c=sg0gIgj-pqT9xZp1nZjt0Q&r=6RVHPGHUyvTPrwD1zOyZf1Sk9IGKRM9jU1kYnWO4h2Y&m=FUlr9yv-3VFrfBFXMXlXh3MsftToUYz3REKZFICta2F1W1NhxXHIgiQ8HBm-wSkL&s=mktHA6NNso7A9kwToX4Tqc3UDYWkts3QSVZBWHIkTbg&e=>,
> or unsubscribe
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AYDV6PAJ4UO3BVLRTCL6ULTWVF35BANCNFSM6AAAAAASQDN67Q&d=DwMCaQ&c=sg0gIgj-pqT9xZp1nZjt0Q&r=6RVHPGHUyvTPrwD1zOyZf1Sk9IGKRM9jU1kYnWO4h2Y&m=FUlr9yv-3VFrfBFXMXlXh3MsftToUYz3REKZFICta2F1W1NhxXHIgiQ8HBm-wSkL&s=6LpxI2MEZtFtlueHZEnNfS9AXEYow032Ygws7rsnuY8&e=>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

*Dominic Vasquez*

IT Systems Engineer, Avant LLC
Avant.com <http://www.avant.com/> | Financially ForwardTM
222 W Merchandise Mart Plaza, Suite 900
Chicago, IL 60654

-- 
This e-mail is only intended for the person(s) to whom it is addressed and 
may contain confidential information.  Any unauthorized review, use, 
disclosure, or distribution is prohibited. If you received this e-mail in 
error, please notify the sender by reply e-mail and then delete this 
message and any attachments from your system. Thank you for your 
cooperation.

**Comment by dvasquezavant** [comment]
Right now using the updated version and installing Ventura for upgrades from Monterey.
Right now not seeing or noticing errors. 
Any other suggestions for me?
Thank you for all

**Comment by dvasquezavant** [comment]
I just received this on one laptop that tried to update. 

**An error occurred preparing the update.    [erase-install] Timeout reached for PID 83508! /Library/Management/erase-install/erase-install.sh: line 2668: 83508 Terminated: 15          /bin/cat 0<&4    [erase-install] Reached end of script. Exit with error 42.     [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success /Library/Management/erase-install/erase-install.sh: line 1046: 56395 Terminated: 15          /usr/bin/caffeinate -dimsu -w $  /Library/Management/erase-install/erase-install.sh: line 1266: kill: (83490) - No such process    [finish] Script exit code: 42 [erase-install-launcher] Exit (42)
--
Error running script: return code was 42.**

**Comment by grahampugh** [comment]
`An error occurred preparing the update.` is output from startosinstall. Not much I can help with there. You could look in `install.log`. Or just try again on that machine.

**Comment by samuelchchan** [comment]
I have exact error code for this issue as well, however, it doesn't seems the issue related to startosinstall. Here is the log

`2023-02-20 23:34:16 [erase-install] v29.0 script execution started: Mon Feb 20 23:34:16 IST 2023
2023-02-20 23:34:16 [erase-install] System version: 12.6 (Build: 21G115)
2023-02-20 23:34:16 [check_for_dialog_app] dialog is installed (/Library/Application Support/Dialog/Dialog.app)
2023-02-20 23:34:16 [erase-install] Caffeinating this script (pid=77685)
2023-02-20 23:34:16 [check_free_space] OK - 61 GB free/purgeable disk space detected
2023-02-20 23:34:16 [erase-install] Looking for existing installer app or pkg
2023-02-20 23:34:16 [find_existing_installer] Installer found at /Applications/Install macOS Ventura.app.
2023-02-20 23:34:16 [check_installer_is_valid] Checking validity of /Applications/Install macOS Ventura.app.
2023-02-20 23:34:16 [check_installer_is_valid] Mounting /Applications/Install macOS Ventura.app/Contents/SharedSupport/SharedSupport.dmg
2023-02-20 23:34:17 [check_installer_is_valid] Using Build value from com_apple_MobileAsset_MacSoftwareUpdate.xml
Volume Shared Support on disk5s2 force-unmounted
2023-02-20 23:34:19 [compare_build_versions] Comparing (1) 21G115 with (2) 22D49
2023-02-20 23:34:19 [check_installer_is_valid] Installer: 22D49 >= System: 21G115 : valid build.
2023-02-20 23:34:19 [erase-install] Checking if the cached installer matches requested build...
2023-02-20 23:34:19 [compare_build_versions] Comparing (1) 22D49 with (2) 22D49
2023-02-20 23:34:19 [compare_build_versions] 22D49 = 22D49
2023-02-20 23:34:19 [erase-install] Existing installer matches requested build.
2023-02-20 23:34:19 [erase-install] Running on architecture arm64
2023-02-20 23:34:19 [get_user_details] ask for user credentials (attempt 1/5)
2023-02-20 23:34:19 [get_default_dialog_args] Invoking utility dialog
2023-02-20 23:34:19.805 Dialog[78143:127274016] XType: failed to connect - Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process." UserInfo={NSDebugDescription=The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process.}
2023-02-20 23:34:19.806 Dialog[78143:127274016] Font server protocol version mismatch (expected:5 got:0), falling back to local fonts
2023-02-20 23:34:19.806 Dialog[78143:127274016] XType: unable to make a connection to the font daemon!
2023-02-20 23:34:19.806 Dialog[78143:127274016] XType: XTFontStaticRegistry is enabled as fontd is not available.
yes rushikadeoras is a member of everyone
2023-02-20 23:34:27 [get_user_details] rushikadeoras is a Volume Owner
2023-02-20 23:34:27 [check_password] Success: the password entered is the correct login password for rushikadeoras.
2023-02-20 23:34:27 [check_power_status] OK - AC power detected
2023-02-20 23:34:27 [erase-install] Installer is at: /Applications/Install macOS Ventura.app
2023-02-20 23:34:27 [check_free_space] OK - 61 GB free/purgeable disk space detected

2023-02-20 23:34:27 [erase-install] WARNING! Running /Applications/Install macOS Ventura.app with reinstall option

2023-02-20 23:34:27 [erase-install] Sending to dialog: quit:
2023-02-20 23:34:27 [get_default_dialog_args] Invoking utility dialog
2023-02-20 23:34:28 [erase-install] Writing LaunchDaemon which will remove /Library/Management/erase-install at next boot
2023-02-20 23:34:28.111 Dialog[79011:127275924] XType: failed to connect - Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process." UserInfo={NSDebugDescription=The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process.}
2023-02-20 23:34:28.111 Dialog[79011:127275924] Font server protocol version mismatch (expected:5 got:0), falling back to local fonts
2023-02-20 23:34:28.111 Dialog[79011:127275924] XType: unable to make a connection to the font daemon!
2023-02-20 23:34:28.111 Dialog[79011:127275924] XType: XTFontStaticRegistry is enabled as fontd is not available.
2023-02-20 23:34:28 [launch_startosinstall] Launching startosinstall
2023-02-20 23:34:28 [erase-install] Sending password to startosinstall
2023-02-20 23:34:39.327 Dialog[79011:127275925] Spell server connection invalidated
[erase-install] Timeout reached for PID 79055!
2023-02-21 00:34:29 [erase-install] Reached end of script. Exit with error 42.

2023-02-21 00:34:29 [erase-install] attempting to terminate the &apos;caffeinate&apos; process - Termination message indicates success

2023-02-21 00:34:29 [finish] quitting dialog
2023-02-21 00:34:30 [finish] Script exit code: 42`

**Comment by reuven-cohen** [comment]
Just ran into the same issue and the solution was staring me in the face.... Check the hardware model and verify it's not out of bounds for macOS Ventura.
We had a mid 2015 MBP that was failing with the same error above. macOS Venutra is for 2017 and above.

https://support.apple.com/en-us/HT213264

**Comment by grahampugh** [comment]
Hmm, erase-install shouldn't allow you to proceed with an incompatible installer. That might be something mist is not as good as detecting as installinstallmacos was but not sure about that.

**Comment by reuven-cohen** [comment]
System Preferences > Software Update shows that it is on the latest supported macOS version 12.6.3.

Im using EraseInstall PKG v28.1 using the command below to run:
`/Library/Management/erase-install/erase-install.sh --reinstall --os=13 --update --min-drive-space=40 --current-user --check-power --no-fs --depnotify --cleanup-after-use`

Below is the log output on the Mid 2015 macBook Pro:

`2023-03-02 11:48:19  [erase-install] Caffeinating this script (pid=4670)
2023-03-02 11:48:20  [check_free_space] OK - 439 GB free/purgeable disk space detected
2023-03-02 11:48:20  [erase-install] Looking for existing installer app or pkg
2023-03-02 11:48:20  [find_existing_installer] Installer found at /Applications/Install macOS Ventura.app.
2023-03-02 11:48:20  [check_installer_is_valid] Checking validity of /Applications/Install macOS Ventura.app.
2023-03-02 11:48:34  [check_installer_is_valid] Mounting /Applications/Install macOS Ventura.app/Contents/SharedSupport/SharedSupport.dmg
2023-03-02 11:48:36  [check_installer_is_valid] Using Build value from com_apple_MobileAsset_MacSoftwareUpdate.xml
Volume Shared Support on disk2s2 force-unmounted
2023-03-02 11:48:37  [compare_build_versions] Comparing (1) 21G419 with (2) 22D68
2023-03-02 11:48:37  [check_installer_is_valid] Installer: 22D68 >= System: 21G419 : valid build.
2023-03-02 11:48:37  [erase-install] Running on architecture i386
2023-03-02 11:48:38  [check_power_status] OK - AC power detected
2023-03-02 11:48:38  [erase-install] Installer is at: /Applications/Install macOS Ventura.app
2023-03-02 11:48:38  [check_free_space] OK - 439 GB free/purgeable disk space detected

2023-03-02 11:48:38  [erase-install] WARNING! Running /Applications/Install macOS Ventura.app with reinstall option

2023-03-02 11:48:38  [erase-install] Sending to dialog: quit:
2023-03-02 11:48:38  [get_default_dialog_args] Invoking utility dialog
2023-03-02 11:48:38  [erase-install] Writing LaunchDaemon which will remove /Library/Management/erase-install at next boot
2023-03-02 11:48:39  [launch_startosinstall] Launching startosinstall
Error: The update cannot be installed on this computer.
By using the agreetolicense option, you are agreeing that you have run this tool with the license only option and have read and agreed to the terms.
If you do not agree, press CTRL-C and cancel this process immediately.
   [erase-install] Timeout reached for PID 4798!
2023-03-02 12:48:41  [erase-install] Reached end of script. Exit with error 42.

2023-03-02 12:48:41  [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success

2023-03-02 12:48:41  [finish] quitting dialog
2023-03-02 12:48:41  [finish] Script exit code: 42
`

**Comment by grahampugh** [comment]
How was the macOS 13 installer downloaded? Using erase-install?

**Comment by reuven-cohen** [comment]
As far as I can tell, it may not have even downloaded the installer but instead just hung after starting EraseInstall then failed with the error above.

**Comment by grahampugh** [comment]
The point is that the installer is already on the disk. The compatibility logic is done by mist at the listing phase, not in the erase-install script, so if you got the installer by some other means, erase-install will try and use it. What I don't expect is that mist would offer that installer for download, but I probably have not tested that as thoroughly as I did with my old installinstallmacos.py script that used to do that check.

**Comment by reuven-cohen** [comment]
I had deleted '/library/management/eraseinstall' then re-ran the installer and command. It still popped up indicating that it was downloading and gave the same error. No ventura installer was present on the machine.

**Comment by grahampugh** [comment]
"It still popped up indicating that it was downloading" - in the last output you posted here, the Installer was already on the machine. Deleting erase-install won't affect that at all - you would have to delete the installer itself (in the Applications folder) to test out whether mist is working as expected.

**Comment by reuven-cohen** [comment]
Looks like "Install macOS Ventura.app" was present on the machine. I will delete and test to see if the error keeps coming up.

**Comment by DaveKozl** [comment]
Hi Graham,

I also get a similar error when trying to upgrade from 12.6.1 to Ventura 13.2.1, the user's password is accepted, but then error 42 pops up, here is a longer log: 

Ventura's image on the station was fetched from your erase-install tool

By the way, I also tested the same installation model on several test stations and I did not notice a problem here, although it looks like a system error to me.

Script result: [erase-install-launcher] Starting script "/Library/Management/erase-install/erase-install.sh" --reinstall --check-power --version=13.2.1

2023-03-21 17:11:40  [erase-install] v29.1 script execution started: Tue Mar 21 17:11:40 CET 2023
2023-03-21 17:11:40  [erase-install] System version: 12.6.1 (Build: 21G217)
2023-03-21 17:11:40  [check_for_dialog_app] dialog is installed (/Library/Application Support/Dialog/Dialog.app)
2023-03-21 17:11:40  [check_for_dialog_app] Creating dialog log (/var/tmp/dialog.sc7)...
2023-03-21 17:11:40  [log_rotate] Start rotating logs in /Library/Management/erase-install/log
2023-03-21 17:11:40  [log_rotate] moving /Library/Management/erase-install/log/erase-install.log.2 to /Library/Management/erase-install/log/erase-install.log.3
2023-03-21 17:11:40  [log_rotate] moving /Library/Management/erase-install/log/erase-install.log.1 to /Library/Management/erase-install/log/erase-install.log.2
2023-03-21 17:11:40  [log_rotate] moving /Library/Management/erase-install/log/erase-install.log to /Library/Management/erase-install/log/erase-install.log.1
2023-03-21 17:11:40  [log_rotate] Finished rotating logs in /Library/Management/erase-install/log
2023-03-21 17:11:40  [erase-install] Caffeinating this script (pid=66061)
2023-03-21 17:11:41  [check_free_space] OK - 830 GB free/purgeable disk space detected
2023-03-21 17:11:41  [erase-install] Looking for existing installer app or pkg
2023-03-21 17:11:41  [find_existing_installer] Installer found at /Applications/Install macOS Ventura.app.
2023-03-21 17:11:41  [check_installer_is_valid] Checking validity of /Applications/Install macOS Ventura.app.
Volume Shared Support on disk5s2 force-unmounted
2023-03-21 17:11:42  [check_installer_is_valid] Mounting /Applications/Install macOS Ventura.app/Contents/SharedSupport/SharedSupport.dmg
2023-03-21 17:11:43  [check_installer_is_valid] Using Build value from com_apple_MobileAsset_MacSoftwareUpdate.xml
Volume Shared Support on disk5s2 force-unmounted
2023-03-21 17:11:45  [compare_build_versions] Comparing (1) 21G217 with (2) 22D68
2023-03-21 17:11:45  [check_installer_is_valid] Installer: 22D68 >= System: 21G217 : valid build.
2023-03-21 17:11:45  [erase-install] Running on architecture arm64
2023-03-21 17:11:45  [get_user_details] ask for user credentials (attempt 1/5)
2023-03-21 17:11:45  [get_default_dialog_args] Invoking utility dialog
yes xxxxx is a member of everyone
2023-03-21 17:11:57  [get_user_details] xxxxx is a Volume Owner
2023-03-21 17:12:00  [check_password] Success: the password entered is the correct login password for xxxxx.
2023-03-21 17:12:00  [check_power_status] WARNING - No AC power detected
2023-03-21 17:12:00  [get_default_dialog_args] Invoking utility dialog
2023-03-21 17:12:18  [check_power_status] OK - AC power detected
2023-03-21 17:12:18  [check_power_status] Sending quit message to dialog log (/var/tmp/dialog.sc7)
2023-03-21 17:12:18  [erase-install] Installer is at: /Applications/Install macOS Ventura.app
2023-03-21 17:12:18  [check_free_space] OK - 830 GB free/purgeable disk space detected

2023-03-21 17:12:18  [erase-install] WARNING! Running /Applications/Install macOS Ventura.app with reinstall option

2023-03-21 17:12:18  [erase-install] Sending quit message to dialog log (/var/tmp/dialog.sc7)
2023-03-21 17:12:18  [get_default_dialog_args] Invoking utility dialog
2023-03-21 17:12:19  [launch_startosinstall] Launching startosinstall
2023-03-21 17:12:19  [erase-install] Sending password to startosinstall
Error: could not get authorization...
By using the agreetolicense option, you are agreeing that you have run this tool with the license only option and have read and agreed to the terms.
If you do not agree, press CTRL-C and cancel this process immediately.
   [erase-install] Timeout reached for PID 67911!
2023-03-21 18:12:19  [erase-install] Reached end of script. Exit with error 42.

2023-03-21 18:12:19  [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success

2023-03-21 18:12:19  [finish] sending quit message to dialog (/var/tmp/dialog.sc7)
2023-03-21 18:12:19  [finish] Script exit code: 42
[erase-install-launcher] Exit (42)

**Comment by kMikaZu** [comment]
Same problem here, where it worked 2 weeks ago just fine. 

- The scripts downloads the installer
- It then starts the upgrade but the progress bar is stuck in "Knight Rider" modus and nothing gets installed. Same error code...

**Comment by grahampugh** [comment]
@DaveKozl your error is different:

```
Error: could not get authorization...
```

Something wrong with your user.

**Comment by grahampugh** [comment]
@kMikaZu the error code isn't enough information. See above for at least three different errors.

**Comment by kMikaZu** [comment]
Ok, a `--fetch-full-installer` option seems to do the trick here. But still strange it worked before. What would be the best option to cache the installer first @grahampugh ?

**Comment by grahampugh** [comment]
Please try v29.2.

**Comment by MehdiYawari** [comment]
Issue still persists:

2023-06-12 17:06:12  [erase-install] v29.2 script execution started: Mo 12 Jun 2023 17:06:12 CEST
2023-06-12 17:06:12  [erase-install] System version: 13.0 (Build: 22A380)
2023-06-12 17:06:12  [check_for_dialog_app] dialog is installed (/Library/Application Support/Dialog/Dialog.app)
2023-06-12 17:06:12  [check_for_dialog_app] Creating dialog log (/var/tmp/dialog.Y18)...
2023-06-12 17:06:12  [log_rotate] Start rotating logs in /Library/Management/erase-install/log
2023-06-12 17:06:12  [log_rotate] Finished rotating logs in /Library/Management/erase-install/log
2023-06-12 17:06:12  [erase-install] Caffeinating this script (pid=19882)
2023-06-12 17:06:12  [check_free_space] OK - 973 GB free/purgeable disk space detected
2023-06-12 17:06:12  [erase-install] Looking for existing installer app or pkg
2023-06-12 17:06:12  [find_existing_installer] Installer found at /Applications/Install macOS Ventura.app.
2023-06-12 17:06:12  [check_installer_is_valid] Checking validity of /Applications/Install macOS Ventura.app.
Volume Shared Support on disk2s2 force-unmounted
2023-06-12 17:06:13  [check_installer_is_valid] Mounting /Applications/Install macOS Ventura.app/Contents/SharedSupport/SharedSupport.dmg
2023-06-12 17:06:14  [check_installer_is_valid] Using Build value from com_apple_MobileAsset_MacSoftwareUpdate.xml
Volume Shared Support on disk2s2 force-unmounted
2023-06-12 17:06:15  [compare_build_versions] Comparing (1) 22A380 with (2) 22F2073
2023-06-12 17:06:15  [check_installer_is_valid] Installer: 22F2073 >= System: 22A380 : valid build.
2023-06-12 17:06:15  [erase-install] Running on architecture i386
2023-06-12 17:06:15  [check_power_status] OK - AC power detected
2023-06-12 17:06:15  [erase-install] Installer is at: /Applications/Install macOS Ventura.app
2023-06-12 17:06:15  [erase-install] Invoking --move option
2023-06-12 17:06:15  [move_to_applications_folder] Valid installer already in /Applications folder
2023-06-12 17:06:15  [check_free_space] OK - 973 GB free/purgeable disk space detected

2023-06-12 17:06:15  [erase-install] WARNING! Running /Applications/Install macOS Ventura.app with eraseinstall option

2023-06-12 17:06:15  [erase-install] Sending quit message to dialog log (/var/tmp/dialog.Y18)
2023-06-12 17:06:15  [get_default_dialog_args] Invoking fullscreen dialog
2023-06-12 17:06:15  [launch_startosinstall] Launching startosinstall
### **Error: Das Update kann auf diesem Computer nicht installiert werden.
By using the agreetolicense option, you are agreeing that you have run this tool with the license only option and have read and agreed to the terms.
If you do not agree, press CTRL-C and cancel this process immediately.**
^C   [terminate] Script was interrupted (last exit code was 130)

2023-06-12 17:06:44  [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success

2023-06-12 17:06:44  [finish] sending quit message to dialog (/var/tmp/dialog.Y18)
2023-06-12 17:06:44  [finish] Script exit code: 143

**Comment by grahampugh** [comment]
@MehdiYawari please don't assume that the reason your installation is failing is the same as the other people in this post, many of whom have posted different error messages. You would all be better to start a new issue to avoid confusion. 

In your case I have to ask again if you have any software restriction or security software that might be interfering with the installer. You should try running startosinstall stand-alone on that client and see if that's working.

**Comment by kMikaZu** [comment]
> 

Same result...

**Comment by kMikaZu** [comment]
> Please try v29.2.

Same result...

**Comment by NikThanki** [comment]
I have started to see the same issue even after using 29.2. I checked the installer that was downloaded and ran it manually.  The installer says it is not compatible with the device I ran the script on (apple silicon). Currently testing using software update on the Mac to pin point where the issue lies.  I have been using your script for many months now with no issues.  I almost feel that it could be something that has changed on the apple side (despite the Ventura installer claiming its is universal) or mist?
